### PR TITLE
EnBili OK and documented

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
AKA Biri.

All functions matched, almost everything named (excluding a couple of stray dLists). Enums for params and limbs. Collider flags put in manually, although they're fairly simple so I think they're all correct.

Simple enemy actor (no actual attacks, only bumper damage), with a couple of strange functions.

I'm prepared to argue about whether they're oral arms or tentacles, by the way: Nintendo don't seem very sure themselves about their jellyfish anatomy.

### CODE REVIEW

**Comment by fig02** [comment]
Even if the name you used is more correct, Id say tentacles will be a million times more recognizable to people reading the code. Have never heard of oral arms before in my life, and im not sure many people have lmfao

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 1]
```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bili/func_809C0600.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bili/func_809C067C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bili/func_809C06E0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Bili/func_809C0754.s")
+static CollisionCheckInfoInit2 sColChkInfoInit = { 1, 9, 28, -20, 30 };
+
+DamageTable sDamageTable = {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c]: static

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 2]
```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
...
+
+void EnBili_Frozen(EnBili* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (!(this->actor.flags & 0x8000)) {
+        this->actor.gravity = -1.0f;
+    }
+
+    if ((this->actor.bgCheckFlags & 1) || (this->actor.floorHeight == -32000.0f)) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:528]: `BGCHECK_Y_MIN` for -32000.0f

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 3]
```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
...
+            if (Actor_ApplyDamage(&this->actor) == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_BIRI_DEAD);
+                func_80032C7C(globalCtx, &this->actor);
+                this->actor.flags &= ~1;
+            }
+
+            damageEffect = this->actor.colChkInfo.damageEffect;
+
+            if (damageEffect == 0x1) { // Deku Nut
+                if (this->actionFunc != EnBili_Stunned) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:552]: you can make an enum for the damage effect if you like. you would use them both in the damage table and in the code

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 4]
```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
...
+        EnBili_PulseLimb4(this, curFrame, &limbScale);
+        rot->y = (Camera_GetCamDirYaw(ACTIVE_CAM) - this->actor.shape.rot.y) + 0x8000;
+    }
+
+    Matrix_Scale(limbScale.x, limbScale.y, limbScale.z, MTXMODE_APPLY);
+    return false;
+}
+
+static u64* sOralArmsTextures[] = { 0x06000E08, 0x06001708, 0x06002008, 0x06002908,
+                                    0x06003208, 0x06003B08, 0x06004408, 0x06004D08 };
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:730]: id put a comma on the last one and reformat

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 5]
```diff
@@ -46,69 +59,705 @@ static ColliderCylinderInit D_809C1640 = {
...
+    this->timer = 300;
+    this->actionFunc = EnBili_Frozen;
+}
+
+// Miscellaneous
+
+/**
+ * Changes the texture displayed on the oral arms limb using the current frame.
+*/
+void EnBili_UpdateOralArmsIndex(EnBili* this) {
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:278]: i still think that tentacles are going to be much easier for the average reader to understand, even if its technically less correct lol

**Review by fig02 — APPROVED** [review]
looks good other than the `oral arms` comment that was ignored rofl :D

File: src/overlays/actors/ovl_En_Bili/z_en_bili.h [hunk 6]
```diff
@@ -6,11 +6,34 @@
...
+    /* 0x01B6 */ Vec3s morphTable[5];
+    /* 0x01D4 */ ColliderCylinder collider;
 } EnBili; // size = 0x0220
 
 extern const ActorInit En_Bili_InitVars;
 
+typedef enum {
+    /* -1 */ EN_BILI_TYPE_NORMAL = -1,
+    /* 0 */ EN_BILI_TYPE_VALI_SPAWNED,
+    /* 1 */ EN_BILI_TYPE_DYING
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.h:28]:
```suggestion
    /*  0 */ EN_BILI_TYPE_VALI_SPAWNED,
    /*  1 */ EN_BILI_TYPE_DYING
```

File: src/overlays/actors/ovl_En_Bili/z_en_bili.h [hunk 7]
```diff
@@ -6,11 +6,34 @@
...
+} EnBiliType;
+
+typedef enum {
+    /* 0 */ EN_BILI_LIMB_NONE,
+    /* 1 */ EN_BILI_LIMB_1,
+    /* 2 */ EN_BILI_LIMB_BELL,
+    /* 3 */ EN_BILI_LIMB_STOMACH,
+    /* 4 */ EN_BILI_LIMB_ORALARMS
+} EnBiliLimbs;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.h:37]:
```suggestion
} EnBiliLimb;
```

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 8]
```diff
@@ -46,69 +59,714 @@ static ColliderCylinderInit D_809C1640 = {
...
+// Action functions
+
+void EnBili_FloatIdle(EnBili* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+
+    if (!(this->timer % 4)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:337]:
```suggestion
    if ((this->timer % 4) == 0) {
```

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c [hunk 9]
```diff
@@ -46,69 +59,714 @@ static ColliderCylinderInit D_809C1640 = {
...
+
+void EnBili_DischargeLightning(EnBili* this, GlobalContext* globalCtx) {
+    static Color_RGBA8 primColor = { 255, 255, 255, 255 };
+    static Color_RGBA8 envColor = { 200, 255, 255, 255 };
+    s32 i;
+    Vec3f effectPos;
+    s16 effectYaw;
+
+    for (i = 0; i < 4; i++) {
+        if (!((this->timer + (i << 1)) % 4)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Bili/z_en_bili.c:370]:
```suggestion
        if (((this->timer + (i << 1)) % 4) == 0) {
```

File: src/overlays/actors/ovl_En_Bili/z_en_bili.c (hunk 8 above)
Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Bili/z_en_bili.c:337]: This doesn't work: the compiler simplifies it into `& 4 == 0`

**Review by Roman971 — APPROVED** [review]

**Comment by fig02** [comment]
discussion about oral arms was still ignored again lmfao, but gonna merge for now and we can bring up again later :)

**Pull request merged**